chain/store.go

```diff
@@ -84,8 +83,13 @@ type Store struct {
 	tipIndex *TipIndex
 }
 
+type ipldStateThing interface {
```

Comment:
I like the idea of this being an interface as it makes the test setup a bit easier, thoughts?